base/pmap.jl

```diff
@@ -0,0 +1,79 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+
+"""
+    pgenerate([::WorkerPool], f, c...) -> iterator
+
+Apply `f` to each element of `c` in parallel using available workers and tasks.
+
+For multiple collection arguments, apply f elementwise.
+
+Results are returned in order as they become available.
+
+Note that `f` must be made available to all worker processes; see
+[Code Availability and Loading Packages](:ref:`Code Availability
+and Loading Packages <man-parallel-computing-code-availability>`)
+for details.
+"""
+function pgenerate(p::WorkerPool, f, c)
+    if length(p) == 0
+        return asyncgenerate(f, c)
+    end
+    batches = batchsplit(c, min_batch_count = length(p) * 3)
+    return flatten(asyncgenerate(remote(p, b -> asyncmap(f, b)), batches))
+end
+
+pgenerate(p::WorkerPool, f, c1, c...) = pgenerate(p, a->f(a...), zip(c1, c...))
+
+pgenerate(f, c) = pgenerate(default_worker_pool(), f, c...)
+pgenerate(f, c1, c...) = pgenerate(a->f(a...), zip(c1, c...))
+
+
+
+"""
+    pmap([::WorkerPool], f, c...) -> collection
+
+Transform collection `c` by applying `f` to each element using available
+workers and tasks.
+
+For multiple collection arguments, apply f elementwise.
+
+Note that `err_retry=true` and `err_stop=false` are deprecated,
+use `pmap(retry(f), c)` or `pmap(@catch(f), c)` instead
+(or to retry on a different worker, use `asyncmap(retry(remote(f)), c)`).
+
+Note that `f` must be made available to all worker processes; see
+[Code Availability and Loading Packages](:ref:`Code Availability
+and Loading Packages <man-parallel-computing-code-availability>`)
+for details.
+"""
+pmap(p::WorkerPool, f, c...) = collect(pgenerate(p, f, c...))
+#pmap(f, c...) = pmap(default_worker_pool(), f, c...)  -- See depricated.jl
```

Comment:
> On the other hand, as it stands, when pmap is removed from deprecated, pmap will stop working completely. Which will be noticed fairly quickly, so it is not too bad as is.

I don't understand. If this definition is in this file and not in deprecated.jl, it's because it should remain supported, right?